facilitator/tests/integration_tests.rs

```diff
@@ -311,9 +311,11 @@ fn aggregation_including_invalid_batch() {
     .unwrap()
     .generate_sum_part(&batch_uuids_and_dates, || {})
     .unwrap_err();
-    // Ideally we would be able to match on a variant in an error enum to check
-    // what the failure was but for now check the error description
-    assert!(err.to_string().contains("packet file digest in header"));
+
+    match err {
+        Error::BadPeerValidationBatch(_) => (),
+        _ => assert!(false, "unexpected error {:?}", err),
+    };
```

Comment:
This works, but I'm also a fan of `assert!(matches!(err, Error::BadPeerValidationBatch(_)))`

Reply:
I've since learned about [`assert_matches`](https://docs.rs/assert_matches/1.5.0/assert_matches/), which is really slick because besides asserting on a pattern match, it also lets you bind parts of the pattern for further matches. e.g.
```
assert_matches!(err, Error::BadPeerValidationBatch(e) => {
    assert_eq!(e, "whatever expected error message");
});
```